docs/concepts/feature-generation.md

```diff
@@ -39,10 +46,22 @@ settings = MaterializationSettings("nycTaxiMaterializationJob",
 client.materialize_features(settings)
 ```
 
-([BackfillTime API doc](https://feathr.readthedocs.io/en/latest/feathr.html#feathr.BackfillTime),
-[client.materialize_features() API doc](https://feathr.readthedocs.io/en/latest/feathr.html#feathr.FeathrClient.materialize_features))
+Note that if you don't have features available in `now`, you'd better specify a `BackfillTime` range where you have features.
+
+Also, Feathr will submit a materialization job for each of the step for performance reasons. I.e. if you have 
+`BackfillTime(start=datetime(2022, 2, 1), end=datetime(2022, 2, 20), step=timedelta(days=1))`, Feathr will submit 20 jobs to run in parallel for maximum performance.
+
+More reference on the APIs:
+
+- [BackfillTime API doc](https://feathr.readthedocs.io/en/latest/feathr.html#feathr.BackfillTime)
+- [client.materialize_features() API doc](https://feathr.readthedocs.io/en/latest/feathr.html#feathr.FeathrClient.materialize_features)
 
-## Consuming the online features
+
+
+## Consuming features in online environment
+
+After the materialization job is finished, we can get the online features by querying the feature name, with the corresponding keys. In the example above, we query the online features called `f_location_avg_fare` and
```

Comment:
1."by querying the `feature table`, corresponding `entity key` and a list of `feature names`" would be more fit the function parameter consequence. 
2. In the example below.

Reply:
fixed

Reply:
fixed